p2p/security/noise/handshake.go

```diff
@@ -82,7 +77,7 @@ func (s *secureSession) runHandshake(ctx context.Context) (err error) {
 	// will be the size of the maximum handshake message for the Noise XX pattern.
 	// Also, since we prefix every noise handshake message with its length, we need to account for
 	// it when we fetch the buffer from the pool
-	maxMsgSize := 2*noise.DH25519.DHLen() + len(payload) + 2*chacha20poly1305.Overhead
+	maxMsgSize := 2*noise.DH25519.DHLen() + 2*chacha20poly1305.Overhead + 1024 /* payload */
```

Comment:
Just wondering if there is a more efficient way of setting length rather than a hard limit here, I don't have a better way readily available, just want to throw some thoughts here to explore the options.

Reply:
Changed to a fixed value of 2048 in 24459c5264da47b2d1431bc06473a13964fe09ca.

It's not even a hard limit, it's just the size we use to get a buffer from the [buffer pool](https://github.com/libp2p/go-buffer-pool). If we get a buffer that's too small, the worst thing that happens is that `append` would allocate a larger buffer.

We can just use a large enough buffer here. What exactly "large enough" means depends on the amount of early data we send, but 2048 should be enough for practically all use cases. And if not, we can live with that one allocation.

Reply:
Thanks for the clarification, that makes sense. Sorry I was not aware that we can do extra allocation if the limit is exceeded.